packages/search/src/util/pagination/content-paginator.ts

```diff
@@ -0,0 +1,48 @@
+import { encode } from "base-64";
+import { IPageResponse } from "./paginator";
+
+interface IContentPageStart {
+  hub: number;
+  ago: number;
```

Comment:
If this applies to enterprise, this should be `portal`, not `ago`.

Reply:
This is directly invoked from the indexer, so AGO only.

Reply:
So, in the scenario where the app running in enterprise (w/ no access to the the Hub API) is paging through search results, what fn would it call?

Reply:
This is a util function to be used by the Hub Indexer only, as it's only relevant when we have to create a paging cursor that points to remaining Portal (AGO) and Hub Index content remaining to be served. In a Portal environment, only portal content is served with pagination already supported by the API.

The scope of this is currently limited to moving over helper functions from the indexer (NodeJS) to Hub.js (see TP [here](https://esriarlington.tpondemand.com/entity/169296-move-pagination-and-sorting-functions-into), but I can certainly revisit this as implementation begins for a more unified content search API.

Reply:
I guess I would argue that if it's to be used by the Hub indexer only, then why move it out of the indexer and into Hub.js? especially considering that these bits will end up in the client Ember app and never be used.

Reply:
It's a valid question. We are currently performing the pre-requisite work for a more unified content search, with Hub.js as the main entryway. Although this particular function may be a bit more specific to Hub Indexer, it should follow similar patterns for pagination that we are developing for other search work that will also be in Hub.js, e.g. user searching.

Reply:
It's good to have the same patterns, but anything that will not be used on the client should not end up in Hub.js. Tree-shaking doesn't appear to be working in our app (and really it's the exception more than the rule for it to work anywhere), so _every_ line of code in Hub.js goes into the Hub application. Worse, because we have very "chunky" lazy loading, it's safe to assume most if not all Hub.js code lands in the initial page load. Specifically, b/c the search package is included by the app itself (and not in a lazy-loaded engine) it absolutely is included in that initial payload.